File: nlox/Lox/NativeFunctions.cs
class ClockFunction : LoxCallable
{
  public int Arity => 0;

  public object? Call(Interpreter interpreter, List<object?> arguments)
  {
    return (double) DateTimeOffset.Now.ToUnixTimeSeconds();
  }

  public override string ToString() => "<native fn>";
}

File: tslox/src/nativefunctions.ts
import Interpreter from "./interpreter";
import { LoxCallable } from "./loxcallable";

export class ClockFunction implements LoxCallable {
  arity = 0;

  call(interpreter: Interpreter, args: unknown[]): unknown {
    return Date.now();
  }

  toString = () => "<native fn>";
}
